copy.go

```diff
@@ -21,6 +23,10 @@ type timespec struct {
 	Ctime time.Time
 }
 
+type onlyWriter struct {
```

Comment:
Sorry I couldn't give feedbacks in one time but, why do we need this type?
Can we just give `*os.File` to `CopyBuffer`?

Reply:
Unfortunately, there is a "feature" in the copyBuffer which prevents that when we add a `os.File` the buffer is not used but it is bypassed to some other "optimization". That optimization is often not working (because of e.g. too old linux kernels) and then falls back to the plain old copy function but with the default buffer size.
So it is needed to wrap the file to prevent any detours and go streight to the copy command with the buffer.

Reply:
If you have a look at this line https://github.com/golang/go/blob/ba0f8ce50fd982fe6e51f77c515660fffdebe40c/src/io/io.go#L409

This is exactly the bypass I'm talking about. Sometimes this is a good thing, but sometimes it isn't (like if your kernel is bellow 5.3).

Reply:
OK, I've got your point. Thank you for clarification.

IMO, it's better to have another option _to use / not to use_ that optimization which depends on the env of the user.

Reply:
Do you think we should add a control flag to tell if we should wrap the `os.File` or do you think we should use the `bufferSize` and use the plain `Copy` in case of bufferSize is 0 and use the `CopyBuffer` with the wrapping in case the buffersize is valid.

Reply:
Because for calling the `io.CopyBuffer` function with a `os.File` as `Writer` will never ever consider the given buffer. So providing a buffer with `os.File` as reader makes actually no sense at all ... never. 
So letting the buffersize be specified AND setting a flag that we don't want to wrap the writer would be quite useless, because then the buffersize has no effect.

Reply:
This would somehow bring back the old way:

```
if copyBuffer > 0 {
    ... io.CopyBuffer(...)
} else {
   .... io.Copy(...)
}
```

Reply:
@otiai10 What do you think?

Reply:
> So letting the buffersize be specified AND setting a flag that we don't want to wrap the writer would be quite useless, because then the buffersize has no effect.

I understand, finally. Thanks. Then here are my points.

1. Users (≠ we, developers) should be able to explicitly set bufsize, and it must be considered if set.
2. Users should be able to explicitly know the optimization by kernel will be disabled when bufsize is set.

Thus, you are right, we should implement something like you call "the old way" 👉 

I'll take over your branch and tweak a bit, if you don't mind.

Reply:
Be my guest. Thank you very much. Appreciate that.